src/Task.cc

```diff
@@ -1114,6 +1114,14 @@ void Task::post_exec_syscall(const std::string& original_exe_file) {
     remote.infallible_syscall(syscall_number_for_arch_prctl(arch()),
                               ARCH_SET_CPUID, 0);
   }
+  if (arch() == aarch64) {
+    if (remote.syscall(syscall_number_for_prctl(remote.task()->arch()),
+                       PR_SET_TSC, PR_TSC_SIGSEGV, 0, 0) != 0) {
```

Comment:
Why aren't we doing this from `set_up_process_arch()` in `Task.cc` where we do it for x86?

And I think we need to change `Task::reenable_cpuid_tsc()` to work for ARM too.

Reply:
> Why aren't we doing this from set_up_process_arch() in Task.cc where we do it for x86?

That's because the arm64 prctl does not preserve this state across an execve.

> And I think we need to change Task::reenable_cpuid_tsc() to work for ARM too.

Done.

Reply:
> That's because the arm64 prctl does not preserve this state across an execve.

Is this intentionally different from x86? Because that difference sounds very unfortunate :-(

Reply:
Yeah, it would have been difficult to follow x86 here because `PR_SET_TSC` on arm64 is 64-bit only and it's not clear what should happen when a 64-bit process execs a 32-bit process which then execs a 64-bit process.

I think of the x86 behavior as a mistake (which was corrected when `ARCH_SET_CPUID` was added), `PR_SET_TSC` is really a property of the code running in a process and not necessarily that of the child processes.